### PR TITLE
 #159965321 Add report article

### PR DESCRIPTION
**Pull request by micahoriaso**
#### What does this PR do?
- Enable users to report articles

#### Description of the task to be completed?
- Give authenticated users the ability to report articles that violate the terms and conditions of Authors' Haven

#### How should this be manually tested?
$ git clone https://github.com/andela/ah-magnificent6.git
$ git checkout ft-report-article-159965321
$ cd ah-magnificent6
$ cp .env-sample .env [and set the environment variables]
$ python3 -m venv venv
$ source venv/bin/activate
(venv) $ pip install -r requirements.txt
(venv) $ python manage.py migrate
(venv) $ python manage.py runserver
Using [Postman](https://www.getpostman.com/) navigate to 
User registration: `POST {your-server-root}:8000/api/user/signup`
```
{
    "email": "author@user.user",
    "username": "author",
    "password": "good123456"
}
```

User Login: `POST {your-server-root}:8000/api/user/login`
```
{
    "email": "author@user.user",
    "password": "good123456"
}
```
Use the token from the response in the login as the authentication header while creating an article.

screen shot 2018-09-12 at 10 54 16

Create article : POST {your-server-root}:8000/api/articles/
```
{
"title": "My Journey to Andela",
 "description": "This article is about how I joined Andela",
"body": "This is my story to Andela"
}
```

```
{
   'text': 'This article is plagiarised'
}
```
Expect this response
```
{
    "Articles": {
        "message": "Your feedback has been recorded. Authors' Haven thanks you for your service."
    }
}
```

Expect this response
```
{
    "Articles": [
        {
            "text": "Blah blah qwerty",
            "article": ""My Journey to Andela",
            "user": "micahgor"
        },
        {
            "id": 3,
            "text": "This article is discriminates",
            "article": "My Journey to Andela",
            "user": "micahgor"
        },
        {
            "text": "This article is extremely inappropriate",
            "article": "My Journey to Andela",
            "user": "micahgor"
        }
    ]
}
```

```
{
   'text': 'This article is highly plagiarised'
}
```
Expect this response
```
{
    "Article": {
        "text": "This article is highly plagiarised",
        "article": "My Journey to Andela",
        "user": "micahgor"
    }
}
```

Expect this response
```
{
    "Articles": {
        "message": "Report was deleted successfully"
    }
}
```

Expect this response
```
{
    "Article": {
        "text": "Blah blah qwerty",
        "article": "My Journey to Andela",
        "user": micahgor
    }
}
```

#### What are the relevant pivotal tracker stories?
[#159965321](https://www.pivotaltracker.com/story/show/159965321)

#### Background Context
The user can only report an article a maximum of five times

### CODE REVIEW

**Comment by Bileonaire** [comment]
Good work. Kindly restrain from revealing the id of the article in the response. Store the article object in the Database as the foreign key but just reveal the title or slug.

**Comment by pngg** [comment]
To second on what Leon has said, could we also return the `username` instead of the `user id` in the response?

**Comment by micahoriaso** [comment]
@Bileonaire and @pngg  your requests have been worked on. Thanks for your feedback.

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -416,3 +417,208 @@ def post(self, request, slug):
...
+    Create a new article report
+    """
+    permission_classes = (IsAuthenticated,)
+    queryset = ArticleReport.objects.all()
+    serializer_class = ArticleReportSerializer
+    renderer_classes = (ArticleJSONRenderer,)
+
+    def list(self, request, slug):
+        """Method for listing all reports."""
+        """Method for reporting an article."""
```
Comment (parseen254) [inline authors/apps/articles/views.py]: @micahoriaso Which of these two docstrings is unnecessary?

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -416,3 +417,208 @@ def post(self, request, slug):
...
+    delete:
+    Delete an article report
+    put:
+    Update an article report
+    """
+    permission_classes = (IsAuthenticated,)
+    serializer_class = ArticleReportSerializer
+    renderer_classes = (ArticleJSONRenderer,)
+
+    def get_object(self, pk):
```
Comment (parseen254) [inline authors/apps/articles/views.py]: @micahoriaso The name for this method is quite ambiguous.

**Review by parseen254 — CHANGES_REQUESTED** [review]

**Review by parseen254 — APPROVED** [review]

**Pull request merged**